### PR TITLE
add configuration option to limit the number of rows per row group

### LINKED ISSUE #375: Control row group size
Hello, I haven't found how to control row group size.

Yes, I can call `Flush`, but how do I know if row group reached limit size (1GB for example) ?

### PR DESCRIPTION
**Pull request by achille-roussel**
Fixes #375 

This PR adds a new `MaxRowsPerRowGroup` writer option to automatically flush row groups when the configured limit of rows per row group has been reached.

### CODE REVIEW

**Review by kevinburkesegment — APPROVED** [review]

**Review by abraithwaite — APPROVED** [review]
:shipit:

File: config.go [hunk 1]
```diff
@@ -397,6 +402,19 @@ func WriteBufferSize(size int) WriterOption {
...
+// produce in each row group.
+//
+// This limit is useful to control size of row groups in both number of rows and
+// byte size. While controlling the byte size of a row group is difficult to
+// achieve with parquet due to column encoding and compression, the number of
+// rows remains a useful proxy.
+//
+// Defaults to unlimited.
+func MaxRowsPerRowGroup(numRows int64) WriterOption {
+	return writerOption(func(config *WriterConfig) { config.MaxRowsPerRowGroup = numRows })
```
Comment (yonesko) [inline config.go:415]: what if numRows <= ?

Comment (achille-roussel) [reply config.go:415]: Do you need this case to be handled?

Comment (yonesko) [reply config.go:415]: No, just accidentally set zero and got infinite loop inside your lib

Comment (yonesko) [reply config.go:415]: How to get permissions to push branches ?

Comment (achille-roussel) [reply config.go:415]: We generally do not give push permissions to external contributors, however you can follow a typical github workflow:
- fork the repository on your own github account
- make the changes to your fork
- open a pull requests against segmentio/parquet-go from your fork

At which point we can review the changes and merge them in once we all agree on the contribution.

Comment (yonesko) [reply config.go:415]: Thanks! 
Offtop: is there a way to copy a parquet file just with another row group limit (max rows per rows group)

Comment (achille-roussel) [reply config.go:415]: Yes, you can configure the writer to generate the new parquet file with a different row group limit and copy rows from the previous file(s) which will rewrite all the rows and apply the new limits.

**Pull request merged**